File: pset6/credit/credit.py
from cs50 import get_int


card = get_int("Number : ")
card_bac = card
odd = 0
even = 0
flag = False
if card in range(0, 1000000000000):
    print("INVALID")
    exit(1)
else:
    while card != 0:
        if flag is False:
            odd += card % 10
            card = int(card / 10)
            flag = True
        else:
            s = card % 10
            s *= 2
            while s != 0:
                even += s % 10
                s = int(s / 10)
            card = int(card / 10)
            flag = False

    sum = odd + even
    if sum % 10 is 0:
        while card_bac > 100:
            card_bac = int(card_bac / 10)
        if card_bac == 34 or card_bac == 37:
            print("AMEX")   # Check or AMEX card
            exit()
        elif card_bac in range(51, 56):
            print("MASTERCARD")  # Check for MASTERCARD
            exit()
        else:
            card_bac = int(card_bac / 10)
            if card_bac is 4:
                print("VISA")   # Check for VISA
                exit()
            else:
                print("INVALID")  # If invalid
                exit(1)
    else:
        print("INVALID")    # If invalid
        exit(1)

File: pset1/credit/credit.c
#include<stdio.h>
#include<cs50.h>

int main(void)
{
    long card = get_long("Number : ");
    long card_bac = card;
    int odd = 0;
    int even = 0;
    bool flag = false;
    if (card < 1000000000000)           //check for correct length of inpur number
    {
        printf("INVALID\n");
    }
    else
    {
        while (card != 0)               //Check for a valid card
        {
            if (flag == false)
            {
                odd += card % 10;
                card /= 10;
                flag = true;
            }
            else
            {
                int s = card % 10;
                s *= 2;
                while (s != 0)
                {
                    even += s % 10;
                    s = s / 10;
                }
                card /= 10;
                flag = false;
            }
        }

        int sum = odd + even;
        if (sum % 10 == 0)
        {
            while (card_bac > 100)
            {
                card_bac /= 10;
            }
            if (card_bac == 34 || card_bac == 37)
            {
                printf("AMEX\n");                           //check for AMEX card
            }
            else if (card_bac >= 51 && card_bac <= 55)
            {
                printf("MASTERCARD\n");                     //check for MASTERCARD
            }
            else
            {
                card_bac /= 10;
                if (card_bac == 4)
                {
                    printf("VISA\n");                       //check for VISA card
                }
                else
                {
                    printf("INVALID\n");                    //case for an invalid type number with correct checksum
                }
            }

        }
        else
        {
            printf("INVALID\n");                            //default condition for card being invalid for checksum
        }
    }

}